core/src/data/function.rs

```diff
@@ -0,0 +1,28 @@
+use {
+    crate::ast::{Expr, OperateFunctionArg},
+    serde::{Deserialize, Serialize},
+};
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct CustomFunction {
+    pub func_name: String,
+    pub args: Option<Vec<OperateFunctionArg>>,
```

Comment:
Would it be enough with `pub args: Vec<OperateFunctionArg>`?
like.. `ADD_ZERO()`, `ADD_ONE(3), `ADD_TWO(1, 2)`
it does not look that the custom function has explicit `None` state.